helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -979,11 +986,26 @@ public boolean[] setChildren(List<String> paths, List<T> records, int options) {
     return set(paths, records, null, null, options);
   }
 
+  /**
+  * async setChildren with TTL
+   */
+  public boolean[] setChildren(List<String> paths, List<T> records, int options, long ttl) {
+    return set(paths, records, null, null, options, ttl);
+  }
+
   /**
    * async set, give up on error other than NoNode
    */
   boolean[] set(List<String> paths, List<T> records, List<List<String>> pathsCreated,
       List<Stat> stats, int options) {
+    return set(paths, records, pathsCreated, stats, options, ZkClient.TTL_NOT_SET);
+  }
+
+  /**
+   * async set with ttl, give up on error other than NoNode
+   */
+  boolean[] set(List<String> paths, List<T> records, List<List<String>> pathsCreated,
```

Comment:
Same comment as above, we should consider clarifying the behavior of TTL param for when node already exists